issuercerts/issuer.go

```diff
@@ -0,0 +1,37 @@
+// Package issuercerts defines types representing a certificate that issue other
+// certificates.
+package issuercerts
+
+import (
+	"crypto/sha256"
+	"crypto/x509"
+	"math/big"
+
+	"github.com/letsencrypt/boulder/core"
+)
+
+type Issuer struct {
+	Cert *x509.Certificate
+}
+type ID int64
+
+// New reads an issuer certificate from a file and returns an Issuer.
+func FromFile(filename string) (*Issuer, error) {
+	cert, err := core.LoadCert(filename)
+	if err != nil {
+		return nil, err
+	}
+	return &Issuer{cert}, nil
+}
+
+func FromCert(cert *x509.Certificate) *Issuer {
+	return &Issuer{cert}
+}
+
+// idForIssuer generates a stable ID for an issuer certificate, based on a hash
+// of the issuer certificate's bytes. This is used for identifying which issuer
+// issued a certificate in the certificateStatus table.
+func (issuer *Issuer) ID() ID {
```

Comment:
Any particular reason to make this a method rather than being a field that is set on construction?

Reply:
Mainly it means that the ID is not mutable outside the package. We could also achieve that by precalculating the ID and then making the `.ID()` method a getter. I skipped that for now because `.ID()` doesn't tend to get called in tight loops but happy to switch it around (or even just make it a field) if you feel like it's better.

Reply:
I don't think you need to change it, the patten of using a method to calculate a thing that never actually changes always just looks a bit weird to me. Given this isn't an expensive operation it should be fine.